### DISCUSSION TITLE
Lesson 6: Mainnet-fork

### DISCUSSION THREAD

**Discussion by 11296925** [smartcontractkit/full-blockchain-solidity-course-py#1534] [Q&A] [answered]
Hello, I receive an error when I try to get the PriceFeed Data while using the mainnet-fork network. Deployment works and interacting with it does work as wel, but when I try to get the price data it runs into the following error.

<img width="728" alt="Error1" src="https://user-images.githubusercontent.com/36366013/172668852-38dba7d2-021a-4098-ab91-4f4e0d7f0427.png">

I don't have this problem when using the rinkeby network or when using other local networks. I also checked the network address in the config file and this is the correct address for retrieving the pricefeed on the mainnet.

I added the line 'print(fund_me.getPrice())' in the deploy.py file to actually call the pricefeed, which makes it run into this error.

<img width="544" alt="Deploypy" src="https://user-images.githubusercontent.com/36366013/172669353-9261e236-aae0-4f41-aff0-8d7cfad5ff85.png">

Does anyone have any idea why I run into this error? Thanks!

**Comment by cromewar** [comment]
Hello @11296925 
Your are most likely using a wrong configuration of mainnet-fork-dev, please delete that network and add it again, also please double check the price feed address to match the repo one.
Cheers

**Reply by 11296925** [reply]
Thanks for your response. Unfortunately this didn't work. I also tried infura instead of alchemy now. I figured out that the problem has something to do with ganache-cli. Retrieving contracts in brownie console is not possible via mainnet-fork, but it is possible with mainnet.

**Reply by cromewar** [reply]
In this case, maybe something is missing with your brownie config, did you add it correctly on that file the mainnet-fork?

**Comment by Softboi2** [comment]
You are doing it the wrong way

**Comment by 11296925** [comment, marked as answer]
I fixed the problem. It turned out that I had to close Ganache completely and then run it. For some reason the local blockchain in ganache is being used when using mainnet-fork and leaving Ganache open, so for this reason it was impossible to read contracts from the mainnet. I still don't get this issue because ganache-cli automatically uses Ganache when it is running right? And I think Ganache is actually meant to be open all the time? @cromewar Do you have any idea why this is the case? Thanks for all the help!

**Reply by cromewar** [reply]
Oh nice, approach.
Well the reason is how brownie listens to the active networks, when you have ganache UI opened, it occupies specific ports brownie looks up for. 
Ganache CLI just run on execution time, and stops working when you finish the deployments and transactions.
